Add reason codes to Close CID capsules and clarify semantics (#169)
tfpauly/quic-proxy: draft-ietf-masque-quic-proxy.md

~~~
{: #fig-capsule-register-target-cid title="Register Target CID Capsule Format"}

Reason:

> : The reason for this registration. This MUST be NONE (0x00).

Reviewer: ```suggestion
: The reason for this registration. This MUST be NONE (0x00), although a different value can be used for future extensions.
```

Author: updated

> : The reason for this registration. This MUST be DEFAULT (0x00).

Reviewer: ```suggestion
: The reason for this registration. This MUST be DEFAULT (0x00), although a different value can be specified by future extensions.
```

Author: updated

paths. Note that initial REGISTER_CLIENT_CID capsules MAY be sent prior to
receiving an HTTP response from the proxy.

If the client receives a VCID it cannot use, it may re-register the same
client CID with a reason code indicating why the previous VCID was unusable.
The proxy SHOULD use this information to select a more suitable VCID. If the

> reason is TOO_SHORT, the proxy SHOULD select a longer VCID. If the reason is

Reviewer 2: Why is this a SHOULD?

Reviewer: How about
```suggestion
reason is TOO_SHORT, the proxy MUST either select a longer VCID or close the registration. If the reason is
```

| scramble       | Reserved (will be used for final version)  | This Document | Section {{scramble-transform}} |
{: #iana-packet-transforms-table title="Initial Packet Transform Names"}

## CID Capsule Reason Codes {#iana-cid-reasons}

This document establishes a new registry, "CID Capsule Reason Codes",
for reason codes used in REGISTER_CLIENT_CID, REGISTER_TARGET_CID,
CLOSE_CLIENT_CID, and CLOSE_TARGET_CID capsules,
in <[](https://www.iana.org/assignments/masque/masque.xhtml)>.
Registrations in this registry are assigned using the
Specification Required policy (Section 4.6 of [IANA-POLICY]).

| Value | Name       | Description                                      | Specification |
|:------|:-----------|:-------------------------------------------------|:--------------|

> | 0x00  | NONE       | Normal operation                                 | This Document |

Reviewer: ```suggestion
| 0x00  | NO_ERROR       | No error, normal operation                                 | This Document |
```

Reviewer: Or DEFAULT

Author: updated

| scramble       | Reserved (will be used for final version)  | This Document | Section {{scramble-transform}} |
{: #iana-packet-transforms-table title="Initial Packet Transform Names"}

## CID Capsule Reason Codes {#iana-cid-reasons}

This document establishes a new registry, "CID Capsule Reason Codes",
for reason codes used in REGISTER_CLIENT_CID, REGISTER_TARGET_CID,
CLOSE_CLIENT_CID, and CLOSE_TARGET_CID capsules,
in <[](https://www.iana.org/assignments/masque/masque.xhtml)>.
Registrations in this registry are assigned using the

> Specification Required policy (Section 4.6 of [IANA-POLICY]).

Reviewer 2: ```suggestion
This registry governs a 62-bit space and operates under the QUIC
registration policy documented in {{Section 22.1 of QUIC}}. This new registry
includes the common set of fields listed in {{Section 22.1.1 of QUIC}}. In
addition to those common fields, all registrations in this registry MUST include
a "Name" field that contains a short name or label for the Reason.

Permanent registrations in this registry are assigned using the Specification
Required policy ({{Section 4.6 of !IANA-POLICY=RFC8126}}), except for values
between 0x00 and 0x3f (in hexadecimal; inclusive), which are assigned using
Standards Action or IESG Approval as defined in {{Sections 4.9 and 4.10 of
IANA-POLICY}}.
```
